packages/jsapi-components/src/spectrum/ComboBox.tsx

```diff
@@ -18,19 +19,57 @@ export function ComboBox(props: ComboBoxProps): JSX.Element {
     ...pickerProps
   } = usePickerProps<ComboBoxProps>(props);
 
+  const isOpenRef = useRef(false);
+  const inputValueRef = useRef('');
+
   const onInputChange = useCallback(
     (value: string) => {
       onInputChangeInternal?.(value);
-      onSearchTextChange(value);
+
+      // Only apply search text if ComboBox is open.
+      if (isOpenRef.current) {
+        onSearchTextChange(value);
+      }
+      // When the ComboBox is closed, `onInputChange` may have been called as a
+      // result of user search input, ComboBox selection, or by selected key
+      // prop changes. We can't determine the source here, so we clear the search
+      // text and store the search value so that the list is unfiltered the next
+      // time the ComboBox is opened. We also store the search value so we can
+      // re-apply it in `onOpenChange` if the ComboBox is opened by user search
+      // input.
+      else {
+        onSearchTextChange('');
+        inputValueRef.current = value;
```

Comment:
Shouldn't this be set in all cases? Everything else looks good.

Reply:
I'm on the fence on this one. I had only put it here since the only consumer should be the `menuTrigger === 'input'` case which should only follow this `else` case. I was trying to conceptually couple them in an attempt to make the code easier to reason about. That said, there's probably not any harm in setting it, and there could be an edge case that hits it I'm unaware of. In general, reasoning about the relationship of this 2 functions is a bit of a pain.